brave/src/main/java/brave/propagation/B3Propagation.java

```diff
@@ -153,8 +165,20 @@ public FactoryBuilder injectFormats(Span.Kind kind, Format format1, Format forma
     @Override public void inject(TraceContext context, R request) {
       Format[] formats = factory.injectFormats;
       if (request instanceof Request) {
-        Span.Kind kind = ((Request) request).spanKind();
-        formats = factory.kindToInjectFormats.get(kind);
+        switch (((Request) request).spanKind()) {
```

Comment:
I'm tempted to do generic foo in `<R> TraceContext.Injector<R> injector(Setter<R, K> setter);` to find out the bound of `R`.. (reflection look up the TypeVariable of the setter) if we could, and found it was a request, then we wouldn't have to do this runtime check!

not tempted now. just a nag for future.

Reply:
done but not with reflection